### PR TITLE
[system test] Metrics collector test frame integration

### PR DESCRIPTION
**Pull request by see-quick**
### Type of change

- Enhancement / new feature
- Refactoring

### Description

This PR adds a metrics collector from the [skodjob](https://skodjob.io/) in our system test module and replaces our current metrics collector implementation within STs.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

File: systemtest/pom.xml [hunk 1]
```diff
@@ -229,6 +229,11 @@
             <groupId>io.fabric8</groupId>
             <artifactId>openshift-model-operatorhub</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.skodjob</groupId>
+            <artifactId>test-frame-metrics-collector</artifactId>
+            <version>0.1.0</version>
```
Comment (kornys) [inline systemtest/pom.xml:235]: use property instead. and is it ok to include directly in systemtest module instead of root?

Comment (see-quick) [reply systemtest/pom.xml:235]: yeah, will update this thanks :). 

> include directly in systemtest module
sure, thanks

**Comment by kornys** [comment]
also if you will merge this please add strimzi operator as adopters to https://github.com/skodjob/test-frame

**Review by im-konge — APPROVED** [review]
LGTM, just the one comment from @kornys 
Thanks

File: systemtest/pom.xml (hunk 1 above)
Comment (see-quick) [reply systemtest/pom.xml]: Updated :)

**Comment by see-quick** [comment]
> also if you will merge this please add strimzi operator as adopters to https://github.com/skodjob/test-frame

Will do it after this is merged.

**Review by henryZrncik — APPROVED** [review]

**Comment by kornys** [comment]
Maybe update to 0.1.1?

**Comment by see-quick** [comment]
> Maybe update to 0.1.1?

We can simply do it in other PR I think, is it already tested via KRaft and regression pipelines? WDYT @im-konge, @Frawless  ?

**Comment by Frawless** [comment]
> > Maybe update to 0.1.1?
> 
> We can simply do it in other PR I think, is it already tested via KRaft and regression pipelines? WDYT @im-konge, @Frawless ?

wasn't 0.1.1 created due to some troubles that affects also these changes?

**Comment by im-konge** [comment]
I would update to `0.1.1` instead of bumping it in different PR

**Comment by see-quick** [comment]
> > > Maybe update to 0.1.1?
> > 
> > 
> > We can simply do it in other PR I think, is it already tested via KRaft and regression pipelines? WDYT @im-konge, @Frawless ?
> 
> wasn't 0.1.1 created due to some troubles that affects also these changes?

Not these, beca

> > > Maybe update to 0.1.1?
> > 
> > 
> > We can simply do it in other PR I think, is it already tested via KRaft and regression pipelines? WDYT @im-konge, @Frawless ?
> 
> wasn't 0.1.1 created due to some troubles that affects also these changes?

It does not affect these changes. It would if you set default timeout configuration :) but I will update to 0.1.1... :)

File: systemtest/src/main/java/io/strimzi/systemtest/utils/specific/MetricsUtils.java [hunk 2]
```diff
@@ -130,7 +129,7 @@ public static void assertMetricResourcesLowerThanOrEqualTo(MetricsCollector coll
     public static void assertMetricResourcesIs(MetricsCollector collector, String kind, Predicate<Double> predicate, String message) {
         String metric = "strimzi_resources\\{kind=\"" + kind + "\",.*}";
         TestUtils.waitFor("metric " + metric + "is " + message, TestConstants.POLL_INTERVAL_FOR_RESOURCE_READINESS, TestConstants.GLOBAL_TIMEOUT_SHORT, () -> {
-            collector.collectMetricsFromPods();
+            collector.collectMetricsFromPods(TestConstants.GLOBAL_TIMEOUT);
```
Comment (Frawless) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/specific/MetricsUtils.java:132]: It is a little bit weird that collecting has timeout higher than dynamic wait. I would expect that collecting will try few times during the dynamic wait.

File: systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java [hunk 3]
```diff
@@ -778,30 +776,28 @@ void setupEnvironment() throws Exception {
...
-            .withComponentName(clusterOperator.getClusterOperatorName())
+            .withComponent(ClusterOperatorMetricsComponent.create(TestConstants.CO_NAMESPACE, clusterOperator.getClusterOperatorName()))
             .build();
 
-        kafkaCollector.collectMetricsFromPods();
-        kafkaExporterCollector.collectMetricsFromPods();
-        clusterOperatorCollector.collectMetricsFromPods();
+        kafkaCollector.collectMetricsFromPods(TestConstants.GLOBAL_TIMEOUT);
+        kafkaExporterCollector.collectMetricsFromPods(TestConstants.GLOBAL_TIMEOUT);
+        clusterOperatorCollector.collectMetricsFromPods(TestConstants.GLOBAL_TIMEOUT);
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java:801]: Wouldn't be enough to have shorter timeout?

File: systemtest/src/main/java/io/strimzi/systemtest/utils/specific/MetricsUtils.java (hunk 2 above)
Comment (see-quick) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/specific/MetricsUtils.java]: updated

**Review by Frawless — APPROVED** [review]

**Pull request merged**